includes/core/server/request/class-refund-charge.php

```diff
@@ -59,6 +60,31 @@ public function set_amount( int $amount ) {
 		$this->set_param( 'amount', $amount );
 	}
 
+	/**
+	 * Sets the reason for the refund.
+	 *
+	 * @param string|null $reason The reason for the refund.
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function set_reason( ?string $reason ) {
+		$this->set_param( 'reason', $reason );
+	}
+
+	/**
+	 * Sets the source for the refund.
+	 *
+	 * @param string $source The reason for the refund.
+	 * @throws Invalid_Request_Parameter_Exception
+	 */
+	public function set_source( string $source ) {
+		$this->set_param(
+			'metadata',
+			[
+				'refund_source' => $source,
+			]
```

Comment:
Would it make sense to use `array_merge` here for better extensibility and avoiding overwrite of existing metadata in the future?

Reply:
Done in 1a011d3cfa8b039c83292afce97f3d95c952a0af